drivers/ethernet/phy/phy_tja1103.c

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_tja1103_phy
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/net/mii.h>
+#include <zephyr/net/mdio.h>
+#include <zephyr/drivers/mdio.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(phy_tja1103, CONFIG_PHY_LOG_LEVEL);
+
+/* PHYs out of reset check retry delay */
+#define TJA1103_PHY_AWAIT_DELAY_POLL_US			15000U
+/* Number of retries for PHYs out of reset check */
+#define TJA1103_PHY_AWAIT_RETRY_COUNT		    200U
+
+/* TJA1103 PHY identifier */
+#define TJA1103_PHY_ID						0x1BB013
+
+/* MMDs */
+#define PMA_STATUS1                 (0x0001U)
```

Comment:
please check https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/mdio.h
All MDIO MMD/register addresses that are not vendor specific should be added directly to that header so that can be reused by other PHY drivers and we don't duplicate these definitions on each new driver (I already did that cleanup once :) ). Any register that is vendor specific, ie. on MMD30/31, can be kept here.

For example, instead of defining here `BT1_PMA_CONTROL_REG_ADR` we can add in net/mdio.h:
```
/* BASE-T1 PMA/PMD control register */
#define MDIO_PMA_PMD_BT1_CTRL		0x0834U

/* BASE-T1 PMA/PMD control register */
/** BASE-T1 master/slave configuration */
#define MDIO_PMA_PMD_BT1_CTRL_CFG_MST	BIT(14)
/* other bitfields here */
```

As a reference for naming, you can follow Linux includes https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/mdio.h#L32 (but not strictly necessary).

Reply:
Tried to follow that as closely as I could.